strawberryfields/decompositions.py

```diff
@@ -416,7 +416,7 @@ def mach_zehnder(m, n, internal_phase, external_phase, nmax):
     BS[m, n] = 1.0j / np.sqrt(2)
     BS[n, m] = 1.0j / np.sqrt(2)
     BS[n, n] = 1.0 / np.sqrt(2)
-    return BS @ Rinternal @ BS @ Rexternal
+    return np.complex128(np.real_if_close(BS @ Rinternal @ BS @ Rexternal))
```

Comment:
Fixes a bug (?) introduced in NumPy 1.19, which results in significant complex rounding error. If not removed, this complex rounding error propagates throughout all the decompositions, producing incorrect results.

Reply:
@nquesada @antalszava @thisac.

Note that this is roughly the same as doing
```suggestion
    return np.round(BS @ Rinternal @ BS @ Rexternal, 14)
```
based on the documentation of [`np.real_if_close()`](https://numpy.org/doc/stable/reference/generated/numpy.real_if_close.html). Should we change it to make the rounding more explicit?

Reply:
I think it's clearer what's happening with the `np.real_if_close` (despite the ugliness of nested functions like this). The `np.round` statement seems a bit more arbitrary at a glance.

Reply:
`tol=14` could be used with `np.real_if_close` (if wanted to make it more explicit)

Reply:
Also, `np.round()` avoids the casting to real, and then back to complex, which adds overhead